### PR TITLE
support for username and encrypted_password / password field for git and hg material

### PR DESCRIPTION
**Pull request by vrushaliwaykole**
**This should not be merged until 19.4.0 is merged as the contract won't be released until then.**

- Document `format_version 5` - support for `username` and `encrypted_password / password` field while specifying git and hg material.
- Updated test cases

GoCD issue for reference: [#6132](https://github.com/gocd/gocd/issues/6132)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -673,6 +687,19 @@ gitMaterial1:
...
+<a name="git-material-update"/>
+
+Since GoCD `>= 19.4.0` you can specify username and password as such
+
+```yaml
+gitMaterial1:
+  git: "http://my.git.repository.com"
+  branch: feature12
+  username: my_username
+  password: "some password"
```
Comment (ketan) [inline README.md]: I think it's probably better that we document `encrypted_password` over `password` :)

**Comment by tomzo** [comment]
The PR says this is documentation only, but please note that this should be unit tested that yaml is actually transformed into json with username and password fields. Somewhere in [here](https://github.com/tomzo/gocd-yaml-config-plugin/blob/master/src/test/java/cd/go/plugin/config/yaml/transforms/MaterialTransformTest.java)

File: README.md (hunk 1 above)
Comment (tomzo) [reply README.md]: Yep. Considering the copy-paste rate from these examples.

Comment (vrushaliwaykole) [reply README.md]: Sure will do that. Also, will update the readme for other materials as well.

**Comment by bdpiparva** [comment]
@tomzo - we have updated the Readme(https://github.com/tomzo/gocd-yaml-config-plugin/pull/119/commits/ede1187bd86be3f527042637eced209bbd73aaa1) for branch attribute for Hg material. 

Attributes `username`, `encrypted_password/password` and `branch` are optional attributes. Users are advised to use them instead of specifying the same in URL

See issue https://github.com/gocd/gocd/issues/6298 for more details.

**Comment by tomzo** [comment]
Looks good to me. 
One more thought - I think it would be worth mentioning that specifying credentials both in attributes and url will result in a validation error.

I'll merge this when 19.4 is out.

**Comment by bdpiparva** [comment]
@tomzo - I will update the ambiguous credentials validation error in the readme and let you know.

> I'll merge this when 19.4 is out.
Since this is the bundled plugin I think we have to merge it before the 19.4.0 release.

**Comment by tomzo** [comment]
> Since this is the bundled plugin I think we have to merge it before the 19.4.0 release.

Usually yes. But only the test code has changed, so previous version will work too.
I don't mind merging earlier, so ping me when you want a new release before 19.4.0.

**Comment by bdpiparva** [comment]
@tomzo - I have added a note for `ambiguous credentials`. If it looks okay to you please create a release for the plugin. I will update the https://github.com/gocd/gocd to bundle the new release of the plugin.

**Comment by bdpiparva** [comment]
@tomzo - I have updated the gocd to have the latest plugins https://github.com/gocd/gocd/pull/6321

**Comment by tomzo** [comment]
@bdpiparva great. Thank you for all the work.

**Pull request merged**